### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by mre**
This includes updating Tokio to 1.x.
The feature flags of Tokio have been changed and `dns` was merged into `net`.

`fast-socks5` was used in two different versions because of a dependency in async-smtp. This gets addressed in https://github.com/async-email/async-smtp/pull/36, but in the meantime we can divert to the fix from the fork at https://github.com/amaurymartiny/async-smtp/tree/am-fast-socks.

### CODE REVIEW

File: core/src/smtp/mod.rs
Comment (mre) [inline core/src/smtp/mod.rs:337]: In case you're wondering why I changed that: `.sample_iter(&Alphanumeric)` returns `u8` now, so we have to convert that to a `char` first to collect it into a String (because u8 doesn't impl Into<String>).
Also the recommended syntax slightly changed according to the [Rust cookbook](https://rust-lang-nursery.github.io/rust-cookbook/algorithms/randomness.html#create-random-passwords-from-a-set-of-alphanumeric-characters).

**Review by amaurymartiny — APPROVED** [review]
cool, thanks a lot!

**Comment by mre** [comment]
Thanks for the super-quick review. If you have some time, could you also create a release so that we can bump the version in https://github.com/lycheeverse/lychee? ❤️

**Comment by mre** [comment]
On second thought, it might not be possible because of the dependency on the fork of async_smtp. 🤔 So let's wait until the upstream PR gets merged I think.

**Comment by amaurymartiny** [comment]
> On second thought, it might not be possible because of the dependency on the fork of async_smtp. 🤔 So let's wait until the upstream PR gets merged I think.

Yeah, I had the same thought for a second, but still wanted to try. OK then, let's wait, i pinged the author of async-smtp.

btw lychee looks great! Just wanted to give a heads up that this library is AGPL-3.0, which might not fit inside MIT projects.

**Pull request merged**
